ledger/ledger-on-sql/src/app/scala/com/daml/ledger/on/sql/SqlLedgerFactory.scala

```diff
@@ -30,17 +30,25 @@ object SqlLedgerFactory extends LedgerFactory[SqlLedgerReaderWriter, ExtraConfig
     config.copy(allowExistingSchemaForIndex = true)
 
   override def owner(
-      initialLedgerId: Option[LedgerId],
-      participantId: ParticipantId,
-      config: ExtraConfig,
+      config: Config[ExtraConfig],
   )(
       implicit executionContext: ExecutionContext,
       materializer: Materializer,
       logCtx: LoggingContext,
   ): ResourceOwner[SqlLedgerReaderWriter] = {
-    val jdbcUrl = config.jdbcUrl.getOrElse {
+    val jdbcUrl = config.extra.jdbcUrl.getOrElse {
       throw new IllegalStateException("No JDBC URL provided.")
     }
-    SqlLedgerReaderWriter.owner(initialLedgerId, participantId, jdbcUrl)
+    SqlLedgerReaderWriter.owner(
+      config.ledgerId,
+      config.participantId,
+      jdbcUrl,
+      timeServiceBackend(config).getOrElse(
+        throw new IllegalStateException(
+          s"${getClass.getSimpleName} should always provide a timeServerBackend but didn't"))
```

Comment:
```suggestion
          s"${getClass.getSimpleName} should always provide a timeServiceBackend but it didn't"))
```

Reply:
That code is not needed anymore but thanks!